### PR TITLE
SIGINT-985: Airgap Refactoring code.

### PR DESCRIPTION
**Pull request by kiransynopsys**

### CODE REVIEW

**Review by kishorikumar — CHANGES_REQUESTED** [review]

File: src/synopsys-action/synopsys-bridge.ts [hunk 1]
```diff
@@ -145,41 +114,27 @@ export class SynopsysBridge {
...
-      if ((await this.checkIfSynopsysBridgeExists(bridgeVersion)) === false) {
-        info('Downloading and configuring Synopsys Bridge')
-        info('Bridge URL is - '.concat(bridgeUrl))
-        const downloadResponse: DownloadFileResponse = await getRemoteFile(tempDir, bridgeUrl)
-        const extractZippedFilePath: string = inputs.SYNOPSYS_BRIDGE_INSTALL_DIRECTORY_KEY || this.getBridgeDefaultPath()
+      if (!(await this.checkIfSynopsysBridgeExists(bridgeVersion))) {
+        info("Downloading and configuring Synopsys Bridge");
+        info("Bridge URL is - ".concat(bridgeUrl));
+        const downloadResponse: DownloadFileResponse = await getRemoteFile(tempDir, bridgeUrl);
+        const extractZippedFilePath: string = SYNOPSYS_BRIDGE_INSTALL_DIRECTORY_KEY || this.getBridgeDefaultPath();
```
[the remark this answers is not in the document]
Comment (kiransynopsys) [reply src/synopsys-action/synopsys-bridge.ts:127]: we have imported inputs and respective using the SYNOPSYS_BRIDGE_INSTALL_DIRECTORY_KEY attribute 
import { BRIDGE_DOWNLOAD_URL, SYNOPSYS_BRIDGE_INSTALL_DIRECTORY_KEY } from "./inputs";

File: src/synopsys-action/synopsys-bridge.ts [hunk 2]
```diff
@@ -349,21 +304,34 @@ export class SynopsysBridge {
...
     } catch (e) {
-      info('Error while reading version file content: '.concat((e as Error).message))
+      info("Error while reading version file content: ".concat((e as Error).message));
+    }
+    return "";
+  }
+
+  async getSynopsysBridgeExecutablePath(): Promise<void> {
+    this.synopsysBridgePath = this.getBridgeDefaultPath();
+    if (SYNOPSYS_BRIDGE_INSTALL_DIRECTORY_KEY) {
```
[the remark this answers is not in the document]
Comment (kiransynopsys) [reply src/synopsys-action/synopsys-bridge.ts:317]: we have imported inputs and respective using the SYNOPSYS_BRIDGE_INSTALL_DIRECTORY_KEY attribute
import { BRIDGE_DOWNLOAD_URL, SYNOPSYS_BRIDGE_INSTALL_DIRECTORY_KEY } from "./inputs";

File: src/synopsys-action/synopsys-bridge.ts [hunk 3]
```diff
@@ -349,21 +304,34 @@ export class SynopsysBridge {
...
+        error("Unable to retrieve the most recent version from Artifactory URL");
       }
     } catch (e) {
-      info('Error while reading version file content: '.concat((e as Error).message))
+      info("Error while reading version file content: ".concat((e as Error).message));
+    }
+    return "";
+  }
+
+  async getSynopsysBridgeExecutablePath(): Promise<void> {
```
[the remark this answers is not in the document]
Comment (kiransynopsys) [reply src/synopsys-action/synopsys-bridge.ts:315]: renamed function.

File: src/synopsys-action/synopsys-bridge.ts
[the remark this answers is not in the document]
Comment (kiransynopsys) [reply src/synopsys-action/synopsys-bridge.ts:1]: tested and validated in both the usecase.

**Review by lsynopsys — APPROVED** [review]

**Review by spurohitsynopsys — APPROVED** [review]

**Pull request merged**